src/celengine/stardb.cpp

```diff
@@ -1360,6 +1352,7 @@ bool StarDatabase::load(istream& in, const fs::path& resourcePath)
                     startPos = next;
                 }
             }
+            AstroObject::autoIndex();
```

Comment:
this is unacceptable, you call the method just for its side effects

Reply:
Ok. So, separate method just for update intetnal counter?

Reply:
> Ok. So, separate method just for update intetnal counter?

nope. you have a function which returns counter and then decrements it. why not to use it in 2 cases above?

Reply:
To save number of used indexes in case if creation of an object would fail. Maybe not a usual case, but anyway...

Reply:
u have 4 billions of indexes...

Reply:
Now, but who knows the future? And there is possible memory saving, if I implement index map with "slabs". I added "Wip" for a reason...